salt/grains/core.py

```diff
@@ -1665,6 +1665,26 @@ def append_domain():
         grain['append_domain'] = __opts__['append_domain']
     return grain
 
+def fqdns():
+    '''
+    Return all FQDNs grains for the system
+    '''
+    # Provides:
+    # fqdns
+
+    grains = {}
+    fqdns = set()
+
+    addresses = salt.utils.network.ip_addrs(include_loopback=False, interface_data=_INTERFACES)
+    addresses.extend(salt.utils.network.ip_addrs6(include_loopback=False, interface_data=_INTERFACES))
+
+    try:
+        fqdns = map(lambda ip: socket.gethostbyaddr(ip)[0], addresses)
```

Comment:
I think it's better to use the previous version where the `try..except` clause was inside the for loop.

Currently, if an exception is produced by `socket.gethostbyaddr` we lose all the other well calculated fqdn as `map()` won't be assigned to `fqdns` due the exception.
 
Using a for loop + `try..except` would allow us to catch exception per IP and will return all other right calculated fqdns.

Reply:
You are right. Fixed